src/App.tsx

```diff
@@ -3,33 +3,72 @@ import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
-type State = {};
+type State = {
+  query: string,
+};
+
+type Movie = {
+  title: string,
+  description: string,
+  imgUrl: string,
+  imdbUrl: string,
+  imdbId: string;
+}
 
 export class App extends React.Component<{}, State> {
-  state: State = {};
+  state: State = {
+    query: '',
+  };
+
+  handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = event.target;
+
+    this.setState({ query: value.toLowerCase() });
+  };
+
+  getVisibleMovies(movies: Movie[]): Movie[] {
+    const { query } = this.state;
+
+    if (query) {
+      return movies.filter(movie => (
+        movie.description.toLowerCase().includes(query)
+        || movie.title.toLowerCase().includes(query)
+      ))
+    } else {
+      return moviesFromServer;
+    }
```

Comment:
else after return is useless.
```suggestion
    }
    
    return moviesFromServer;
```